### PR TITLE
[EBPF] Minor fixes for local KMT on macOS

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Fixes some issues when running local KMT on macOS

### CODE REVIEW

**Review by usamasaqib — APPROVED** [review]
Can you also fix the argument name in `kmt.ssh-config`. `stacks` => `stack`.

**Comment by gjulianm** [comment]
> Can you also fix the argument name in `kmt.ssh-config`. `stacks` => `stack`.

The issue there is that that argument supports multiple stacks, so changing it would make it inconsistent. I changed the documentation so that the default (which is all stacks) are exported in the SSH config, I think that will make it less confusing for users.

**Comment by brycekahle** [comment]
> The issue there is that that argument supports multiple stacks

What do you think about making it `stack` but iterable? https://docs.pyinvoke.org/en/stable/concepts/invoking-tasks.html#iterable-flag-values

File: tasks/kernel_matrix_testing/compiler.py [hunk 1]
```diff
@@ -32,6 +33,21 @@ def get_build_image_suffix_and_version() -> tuple[str, str]:
...
+        return False
+
+    try:
+        with open(docker_config) as f:
+            config = json.load(f)
+    except json.JSONDecodeError:
+        # Invalid JSON (or empty file), we don't have the helper
+        return False
+
+    return "ddtool" in config.get("credHelpers", {}).values()
```
Comment (brycekahle) [inline tasks/kernel_matrix_testing/compiler.py:48]: FWIW, the ECR registries use `ecr-login` as the cred helper.

Comment (gjulianm) [reply tasks/kernel_matrix_testing/compiler.py:48]: Added that too for the detection. For me  `ddtool auth helpers install` only added the `ddtool` helpers but maybe that's changed recently.

**Comment by gjulianm** [comment]
> > The issue there is that that argument supports multiple stacks
> 
> What do you think about making it `stack` but iterable? [docs.pyinvoke.org/en/stable/concepts/invoking-tasks.html](https://docs.pyinvoke.org/en/stable/concepts/invoking-tasks.html#iterable-flag-values)

That looks like the best solution, yes, changed it. Thx!

**Pull request merged**